core-plugins/src/main/java/co/cask/hydrator/plugin/batch/aggregator/GroupByAggregator.java

```diff
@@ -74,6 +81,29 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     stageConfigurer.setOutputSchema(getOutputSchema(inputSchema, groupByFields, aggregates));
   }
 
+  @Override
+  public void prepareRun(BatchAggregatorContext context) throws Exception {
+    super.prepareRun(context);
+    List<String> inputFields = new ArrayList<>();
+    List<String> outputFields = new ArrayList<>();
+    Schema inputSchema = context.getInputSchema();
+    if (SchemaValidator.validateSchemaIsNotNull(inputSchema, "input")) {
+      //noinspection ConstantConditions
+      inputFields = inputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
+    }
+    Schema outputSchema = context.getOutputSchema();
+    if (SchemaValidator.validateSchemaIsNotNull(outputSchema, "output")) {
+      //noinspection ConstantConditions
+      outputFields = outputSchema.getFields().stream().map(Schema.Field::getName).collect(Collectors.toList());
```

Comment:
group by should not be treated like javascript and python. There should be one transform operation per aggregate configured in the plugin.

For example, if we generate a new field 'avgPrice' by calculating 'avg(price)', that should be a single transform operation where the input is 'price', the output is 'avgPrice', and the operation name is probably something like 'GroupBy:avg'.